005-http-bridge-metrics.md

```diff
@@ -0,0 +1,126 @@
+
+I also already have a Grafana dashboard which shows these metrics.
+
+The JMX-Prometheus collector is the official one provided by Prometheus project and it's also the one used internally by the [JMX Exporter](https://github.com/prometheus/jmx_exporter) which Strimzi is using as an "agent" for exposing all the Kafka metrics to Prometheus.
+It means that it uses a JMX Exporter configuration (YAML) file for mapping (filtering, relabeling, hiding ...) from JMX to Prometheus.
+
+For the `Kafka` custom resource we already have such a configuration file in the `metrics` section which is used to enable/disable and configure metrics.
+There is no such integration in the `KafkaBridge` custom resource right now.
+
+Leaving the way bridge metrics are exposed right now means:
+
+* no HTTP server metrics are exposed via JMX. There is no way to filter, relabeling, hiding them; no way for customization.
+* the `metrics` field to add in the `KafkaBridge` custom resource is going to configure the Kafka clients metrics only which could create confusion for the users.
+* JMXTrans integration cannot leverage on bridge HTTP server metrics but only on the Kafka ones.
+
+## Metrics to expose
+
+### HTTP server metrics
+
+The HTTP metrics could be:
+
+* The number of opened HTTP connections to the bridge.
+* Requests being processed.
+* Requests processed/secs grouped by HTTP method.
+* The requests rate in total and even grouped by response codes (2XX, 4XX, 5XX).
+* Bytes received /sec and Bytes sent /sec.
+
+Furthermore, the HTTP requests/sec metrics coming from Vert.x bring labels like `code` (HTTP response code), `local` (bridge server listening), `remote` (client issuing the request), `method`, and `path` that can be useful for some filtering to define following metrics:
+
+* SEND to `/topics/<topic>`
+* SEND TO PARTITIONS to `/topics/<topic>/partitions/<partition>`
+* CREATE CONSUMER to `/consumers/<group>`
+* SUBSCRIBE to `/consumers/<group>/instances/<consumer>/subscription`
+* DELETE CONSUMER to `/consumers/<group>/instances/<consumer>`
+* POLL to `/consumers/<group>/instances/<consumer>/records`
+* COMMIT to `/consumers/<group>/instances/<consumer>/offsets`
+
+Even JVM and GC metrics are exposed out-of-box.
+
+### Kafka clients metrics
+
+Useful metrics about Kafka part could be the following.
+
+* Number of consumers, producers and related opened connections
+* Producer:
+    * The average number of records sent per second (grouped by topic)
+    * The number of outgoing bytes sent to all brokers per second (grouped by topic)
+    * The averate number of records per second that resulted in errors (grouped by topic)
+* Consumer:
+    * The average number of records consumed per second (grouped by clientId-topic)
+    * The average number of bytes consumed per second (grouped by clientId-topic)
+    * Partitions assigned (grouped by clientId)
```

Comment:
Should we also expose some error metrics such as failed connection attempts etc? Or do you think that it is not needed?